### ISSUE TITLE
inconclusive readme

### ISSUE THREAD

**Issue by roninkaizen** [wmitsuda/otterscan#12]
readme contains not all information needed,
the cross-reference between the erigon-version and the corresponding rpc seems logic,
but does not work from experience-
so the `git remote add otterscan git@github.com:wmitsuda/erigon.git` brought errors-
any ideas?

**Comment by wmitsuda** [comment]
Which errors are you seeing?

**Comment by pawnbot** [comment]
I'd also love installation readme for total noobs with clean environment :)
Say for command above it should be:
`git remote add otterscan https://github.com/wmitsuda/erigon.git`
In Dockerfile I had to replace `"/otterscan-build"` with `"/otterscan-build/"` to make docker build work.

For me everything seems running erigon/rpcdaemon/otterscan
Docker logs doesn't show any error but at the same time it's clearly not connected to rpcdaemon in my case since it doesn't show last block/any transactions.

**Comment by wmitsuda** [comment]
The Dockerfile in the develop branch was under development, you might have checked out an unstable revision.

Today's tag contains a working Dockerfile, please try it.

Regarding the `git remote`, using the git via ssh protocol works for me, what did you get?

**Comment by pawnbot** [comment]
Thank you for quick reply!
> The Dockerfile in the develop branch was under development, you might have checked out an unstable revision.
> Today's tag contains a working Dockerfile, please try it.
Same error by default in clean environment with `v2021.07.02-otterscan`:
```
Step 3/17 : COPY ["package.json", "package-lock.json", "/otterscan-build"]
When using COPY with more than one source file, the destination must be a directory and end with a /
```
Though changing to  "/otterscan-build" resolves it.

> Regarding the git remote, using the git via ssh protocol works for me, what did you get?
For me it wasn't able to fetch remote branch (but I again started from clean environment so didn't have my github keys attached)

Do you have any ideas on best way to debug missing connection between otterscan and rpcdaemon?
I've tried hitting rpcdaemon locally with requests and it returns correct response to all of them.

**Comment by wmitsuda** [comment]
I think I now understand what is happening, my docker installation is configured to build images using buildkit by default, it works fine, look:

```

> otterscan@0.1.0 build-docker
> docker build -t otterscan -f Dockerfile .

[+] Building 365.2s (22/22) FINISHED
 => [internal] load build definition from Dockerfile                                                                         0.1s
 => => transferring dockerfile: 753B                                                                                         0.0s
 => [internal] load .dockerignore                                                                                            0.1s
 => => transferring context: 137B                                                                                            0.0s
 => [internal] load metadata for docker.io/library/nginx:1.21.1-alpine                                                       0.0s
 => [internal] load metadata for docker.io/library/node:12.22.3-alpine                                                       2.6s
 => [auth] library/node:pull token for registry-1.docker.io                                                                  0.0s
 => [stage-1 1/8] FROM docker.io/library/nginx:1.21.1-alpine                                                                 0.0s
 => [internal] load build context                                                                                           55.8s
 => => transferring context: 158.59MB                                                                                       55.4s
 => [builder 1/8] FROM docker.io/library/node:12.22.3-alpine@sha256:05328aa871f251b0643b8dcaa01034f126a8343b51ef302de207ad  12.0s
 => => resolve docker.io/library/node:12.22.3-alpine@sha256:05328aa871f251b0643b8dcaa01034f126a8343b51ef302de207ad153a6b623  0.0s
 => => sha256:ddad3d7c1e96adf9153f8921a7c9790f880a390163df453be1566e9ef0d546e0 2.82MB / 2.82MB                               1.2s
 => => sha256:8bb195d94de45050928c69d40bf7912137c9e99b24a8d50de74451be24bc89fe 24.60MB / 24.60MB                             8.2s
 => => sha256:be9b379f7fa3622fff221bc37f0138ad0e2f62f4b112940a8d9c3dacafa646dd 2.24MB / 2.24MB                               2.4s
 => => sha256:05328aa871f251b0643b8dcaa01034f126a8343b51ef302de207ad153a6b6232 1.43kB / 1.43kB                               0.0s
 => => sha256:6f66d017d5d282adefc96c17d2dec8809fa8e7ce63ebfdfb2af6589c90df688c 1.16kB / 1.16kB                               0.0s
 => => sha256:cfc7e7f8c999ee171a9d7d2e192ee2aba91fa767f83fd17a3fdab6ce3e4319fb 6.53kB / 6.53kB                               0.0s
 => => extracting sha256:ddad3d7c1e96adf9153f8921a7c9790f880a390163df453be1566e9ef0d546e0                                    0.4s
 => => sha256:3b8a9bc999d9d6e93526e2c11dff4ce0d2636868002ac9f09de3a1b8183b4829 281B / 281B                                   1.5s
 => => extracting sha256:8bb195d94de45050928c69d40bf7912137c9e99b24a8d50de74451be24bc89fe                                    2.8s
 => => extracting sha256:be9b379f7fa3622fff221bc37f0138ad0e2f62f4b112940a8d9c3dacafa646dd                                    0.3s
 => => extracting sha256:3b8a9bc999d9d6e93526e2c11dff4ce0d2636868002ac9f09de3a1b8183b4829                                    0.0s
 => [stage-1 2/8] RUN apk add jq                                                                                             9.9s
 => [builder 2/8] WORKDIR /otterscan-build                                                                                   0.9s
 => [builder 3/8] COPY [package.json, package-lock.json, /otterscan-build]                                                   1.8s
 => [stage-1 3/8] COPY 4bytes/signatures /usr/share/nginx/html/signatures/                                                  77.2s
 => [builder 4/8] RUN npm install                                                                                          208.0s
 => [stage-1 4/8] COPY trustwallet/blockchains/ethereum/assets /usr/share/nginx/html/assets/                                 6.0s
 => [stage-1 5/8] COPY nginx.conf /etc/nginx/conf.d/default.conf                                                             0.0s
 => [builder 5/8] COPY [run-nginx.sh, tsconfig.json, craco.config.js, tailwind.config.js, /otterscan-build]                  0.0s
 => [builder 6/8] COPY [public, /otterscan-build/public]                                                                     0.0s
 => [builder 7/8] COPY [src, /otterscan-build/src]                                                                           0.0s
 => [builder 8/8] RUN npm run build                                                                                         36.8s
 => [stage-1 6/8] COPY --from=builder /otterscan-build/build /usr/share/nginx/html/                                          0.1s
 => [stage-1 7/8] COPY --from=builder /otterscan-build/run-nginx.sh /                                                        0.1s
 => exporting to image                                                                                                      58.8s
 => => exporting layers                                                                                                     58.7s
 => => writing image sha256:5b08008b1c01ba862640e6f0b4e90c8b0f91545ccb7d0747c77c4a3ba24c28af                                 0.0s
 => => naming to docker.io/library/otterscan                                                                                 0.0s
```

If I turn off buildkit, it gives me a similar error as yours:

```

> otterscan@0.1.0 build-docker
> docker build -t otterscan -f Dockerfile .

Sending build context to Docker daemon  434.7MB
Step 1/17 : FROM node:12.22.3-alpine AS builder
12.22.3-alpine: Pulling from library/node
ddad3d7c1e96: Already exists
8bb195d94de4: Already exists
be9b379f7fa3: Already exists
3b8a9bc999d9: Already exists
Digest: sha256:05328aa871f251b0643b8dcaa01034f126a8343b51ef302de207ad153a6b6232
Status: Downloaded newer image for node:12.22.3-alpine
 ---> cfc7e7f8c999
Step 2/17 : WORKDIR /otterscan-build
 ---> Running in 3ff3e97cce67
Removing intermediate container 3ff3e97cce67
 ---> c21380136acb
Step 3/17 : COPY ["package.json", "package-lock.json", "/otterscan-build"]
When using COPY with more than one source file, the destination must be a directory and end with a /
```

and you are right, if the `dest` parameter of COPY command is a directory it should end with a /, it just happened that buildkit does not complain about it. Since the docs states so (https://docs.docker.com/engine/reference/builder/#copy), I'll fix the Dockerfile.

It should work now for your installation, but I highly recommend everyone to use buildkit, I'll add a flag to the build script to activate it by default as explained in the docs (https://docs.docker.com/develop/develop-images/build_enhancements/#to-enable-buildkit-builds).

It is very fast, build layers in parallel, and much smarter when sending content to the build context, now I understand why some people has been complaining the build was so slow even after fixing the Dockerfile 😂

I gave up after a few minutes trying to build it with the default legacy build, have no idea how much time it should take, and it should output a much bigger image, the buildkit builder can compress the 1GB 4bytes layer into 8MB, not worth it not to use buildkit.

I tagged a hotfix, can you please try out the tag `v2021.07.02-1-otterscan`?

**Comment by wmitsuda** [comment]
> 
> > Regarding the git remote, using the git via ssh protocol works for me, what did you get?
> 
> For me it wasn't able to fetch remote branch (but I again started from clean environment so didn't have my github keys attached)

I changed the instruction in the README, I saw some complaints in another issue about ssh, so I think it is better to use https then. Thanks for the suggestion.

> 
> Do you have any ideas on best way to debug missing connection between otterscan and rpcdaemon?
> I've tried hitting rpcdaemon locally with requests and it returns correct response to all of them.

I'm working in ways to signal in the UI what is the connection status between Otterscan <-> Erigon, so it is easier to spot config mistakes. Please follow https://github.com/wmitsuda/otterscan/issues/9

**Comment by pawnbot** [comment]
Thank you for update!
Indeed with build kit I can build it straight out of the box and it's way faster.
Sadly for some reason still can't make it to find rpcdaemon, tried with
`docker run --rm -p 80:80 --name otterscan -d otterscan:latest`
and 
`docker run --rm --network="host" --name otterscan -d otterscan:latest`

Again totally noob question do I need to setup specific firewall rules so `http://127.0.0.1:8545` is reachable by Erigon?

**Comment by wmitsuda** [comment]
No need to firewall rules, it tries to connect to erigon at browser level, the container itself does not need outbound connections.

Are you able to curl http://127.0.0.1:8545 ? At least to see if the rpcdaemon process is responding?

**Comment by pawnbot** [comment]
Makes total sense, I was running Erigon/RPCdaemon/Otterscan on my remote node so it's clear why it wasn't able to connect.

**Comment by wmitsuda** [comment]
Good to know! I'll closed this issue because it is becoming too generic and most problems were tackled.

@pawnbot and @roninkaizen, feel free to open more specific issues based on latest release.
